File: Data Structures & Algorithms/rotate-array/submission-0.java
class Solution {
    public void rotate(int[] arr, int k) {
        int n = arr.length;
        k = k % n;
        reverse(arr,n-k,n-1);
        reverse(arr,0,n-k-1);
        reverse(arr,0,n-1);


    }

    public void reverse(int arr[] ,int start,int end){
        while(start<end){
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;

            start++;
            end--;
        }
    }
}

File: Data Structures & Algorithms/kth-distinct-string-in-an-array/submission-0.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        mp = {}
        for i in range(len(arr)):
            mp[arr[i]] = mp.get(arr[i],0)+1
            l = []
        for i in mp:
            if mp.get(i)==1:
                l.append(i)
        
        if k>len(l):
            return ""
        else:
            return l[k-1]
        

